### PR TITLE
kevin's fixes for combining builder

### PR DESCRIPTION
**Pull request by kevmoo**
Some things I discussed w/ @natebosch Friday night...

This is a PR into Gary's branch. Didn't want to just jump in there without notice..

### CODE REVIEW

**Comment by kevmoo** [comment]
@grouma – feel free to merge it in if you approve.

I think we should keep cranking on this branch until this is fully implemented. We should add an example directory to the mono repo and actually exercise it!

File: source_gen/lib/builder.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    // Pattern used for `findAssets`, which must be glob-compatible
+    var pattern = buildStep.inputId.changeExtension('.*$_partFiles').path;
+
+    var inputBaseName =
+        p.basenameWithoutExtension(buildStep.inputId.pathSegments.last);
+
+    // Pattern used to ensure items are only considered if they match
+    // [file name without extension].[valid part id].[part file extension]
+    var restrictedPattern = new RegExp([
+      '^', // start of string
```
Comment (natebosch) [inline source_gen/lib/builder.dart]: do we want this file to be a regex tutorial?

**Comment by kevmoo** [comment]
@grouma – PTAL...

File: source_gen/lib/builder.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    ].join(''));
+
+    var assets = await buildStep
+        .findAssets(new Glob(pattern))
+        .where((id) => restrictedPattern.hasMatch(id.pathSegments.last))
+        .transform(concurrentAsyncMap(buildStep.readAsString))
+        .join('\n');
+    if (assets.isEmpty) return;
+    var partOf = nameOfPartial(await buildStep.inputLibrary, buildStep.inputId);
+    var output = '''
```
Comment (natebosch) [inline source_gen/lib/builder.dart]: [nit] indents nicer as:

```dart
var output = '$defaultFileHeader\n\n'
    'part of $partOf;\n\n'
    '$assets';
```

File: source_gen/lib/builder.dart (hunk 1 above)
Comment (kevmoo) [reply source_gen/lib/builder.dart]: No. Just readable for the next maintainer...

File: source_gen/lib/builder.dart (hunk 2 above)
Comment (kevmoo) [reply source_gen/lib/builder.dart]: So you think...

**Pull request merged**